pkg/controller/controller_binding.go

```diff
@@ -188,9 +211,9 @@ func (c *controller) reconcileServiceInstanceCredential(binding *v1alpha1.Servic
 			errorNonbindableServiceClassReason,
 			s,
 		)
-		c.updateServiceInstanceCredentialStatus(toUpdate)
+		c.setServiceInstanceCredentialForCompletedChange(toUpdate)
 		c.recorder.Event(binding, api.EventTypeWarning, errorNonbindableServiceClassReason, s)
-		return nil
+		return true, errors.New(s)
```

Comment:
We should not return an error here, because we do not want to retry.

Reply:
The way this PR touches the worker, the error return value does not affect whether you retry or not (unless the max retry amount is exceeded). The `completed` bool value is in charge of this now.